components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +378,186 @@ public void setAlias(String alias) {
+                        }
+                        this.certificateInfoArray = certificateInfo;
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Only one certificate has been found");
+                        }
+                        JSONObject obj1 = (JSONObject) jsonCertificateInfoArray.get(0);
+                        CertificateInfo[] certificateInfo = {new CertificateInfo()};
+                        certificateInfo[0].setThumbPrint(IdentityApplicationManagementUtil.generateThumbPrint(
+                                obj1.getString(THUMB_PRINT)));
+                        certificateInfo[0].setCertValue(obj1.getString(CERT_VALUE));
+                        this.certificateInfoArray = certificateInfo;
+                    }
+                }
+                // handle the certificates which are not in json format.
+                else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Handling file based certificate configuration.");
+                    }
+                    // Handle file based certificate configuration. File based configurations are in plain text.
+                    if (certificateValue.contains(IdentityApplicationConstants.END_CERTIFICATE + "\n" +
+                            IdentityApplicationConstants.BEGIN_CERTIFICATE)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("More than one plain text certificate has been found");
+                        }
+                        int numberOfCertificates = StringUtils.countMatches(certificateValue,
+                                IdentityApplicationConstants.END_CERTIFICATE
+                                + "\n" + IdentityApplicationConstants.BEGIN_CERTIFICATE) + 1;
+                        CertificateInfo[] certificateInfo = new CertificateInfo[numberOfCertificates];
+
+                        for (int i = 1; i <= numberOfCertificates; i++) {
+                            certificateInfo[i-1] = new CertificateInfo();
+                            String certificateVal;
+                            if (i == numberOfCertificates) {
+                                certificateVal = certificateValue.substring(StringUtils.ordinalIndexOf(
+                                        certificateValue, IdentityApplicationConstants.BEGIN_CERTIFICATE, i));
+                            } else {
+                                // Get the i+2'th occurrence of BEGIN_CERTIFICATE.
+                                certificateVal = certificateValue.substring(StringUtils.ordinalIndexOf(
+                                        certificateValue, IdentityApplicationConstants.BEGIN_CERTIFICATE, i),
+                                        StringUtils.ordinalIndexOf(
+                                        certificateValue, IdentityApplicationConstants.BEGIN_CERTIFICATE, i + 1));
```

Comment:
from here you need to select the content between -----BEGIN CERTIFICATE---- and -----END CERTIFICATE----- texts.
I think if you can select the text between the i th -----BEGIN CERTIFICATE---- and -----END CERTIFICATE-----, rather than going for i+1 th -----BEGIN CERTIFICATE---- it would ease the logic.